### PR TITLE
Support for Wechat applet

### PR DESCRIPTION
**Pull request by einsqing**
Support for Wechat applet，https://developers.weixin.qq.com/miniprogram/en/dev/

### CODE REVIEW

File: src/RESTController.js [hunk 1]
```diff
@@ -38,6 +38,58 @@ if (typeof XMLHttpRequest !== 'undefined') {
 if (process.env.PARSE_BUILD === 'node') {
   XHR = require('xmlhttprequest').XMLHttpRequest;
 }
+if (process.env.PARSE_BUILD === 'weapp') {
+  class WeappXhr {
```
Comment (dplewis) [inline src/RESTController.js]: Move this to a new file `WeAppXHR.js`? 
Why can't you use XHR? Are there some restrictions?

File: src/Parse.js [hunk 2]
```diff
@@ -267,6 +267,10 @@ if (process.env.PARSE_BUILD === 'node') {
   Parse.Hooks = require('./ParseHooks');
 }
 
+if (process.env.PARSE_BUILD === 'weapp') {
+  wx.Parse = Parse;
```
Comment (dplewis) [inline src/Parse.js]: Parse is already global, would you need this?

File: src/StorageController.weapp.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow
+ */
+localStorage = {
```
Comment (dplewis) [inline src/StorageController.weapp.js]: Is there a reason why you are overriding localStorage? Can you use wx.getStorageSync and others directly below?

File: src/RESTController.js (hunk 1 above)
Comment (einsqing) [reply src/RESTController.js]: The Wechat miniprogram does not support XHR

File: src/Parse.js (hunk 2 above)
Comment (einsqing) [reply src/Parse.js]: The Wechat miniprogram does not support global，And use wx instead of global

File: src/StorageController.weapp.js (hunk 3 above)
Comment (einsqing) [reply src/StorageController.weapp.js]: The Wechat miniprogram does not support localStorage，https://developers.weixin.qq.com/miniprogram/en/dev/api/storage/wx.setStorageSync.html

Comment (dplewis) [reply src/StorageController.weapp.js]: You can remove this and just use wx directly no? It’s basically a wrapper.

File: src/Parse.js (hunk 2 above)
Comment (dplewis) [reply src/Parse.js]: That’s ok as long as wx doesn’t use export a Parse Class in the future, you could override it.. Would wx._Parse make more sense? It’s up to you if you want to keep it.

File: src/RESTController.js (hunk 1 above)
Comment (dplewis) [reply src/RESTController.js]: Do queries etc work with this wx.request? If so you can ignore my comment.

Comment (dplewis) [reply src/RESTController.js]: I see they have a web socket api, you could implement that as well. Do files upload correctly?

Comment (einsqing) [reply src/RESTController.js]: Supporting queries, socket with my support when I have time

File: src/Parse.js (hunk 2 above)
Comment (einsqing) [reply src/Parse.js]: It's customary to use wx.Parse in this way, or use export a Parse。

File: src/RESTController.js (hunk 1 above)
Comment (einsqing) [reply src/RESTController.js]: Livequery has been supported

**Comment by dplewis** [comment]
@einsqing I updated the readme can you have a look? Also if you have time, you can add LocalDatastore support as well.

**Comment by einsqing** [comment]
@dplewis LocalDatastore has been supported

**Comment by dplewis** [comment]
Is there anything you want to add or change in the readme like documentation or usage with WeChat?

**Comment by einsqing** [comment]
@dplewis Wechat minprogram is limited to 2m in size. Generally, we use dist/parse.weapp.min.js directly. You can also require ("parse/weapp"), but you need to turn on building NPM packages

image

**Comment by dplewis** [comment]
Since you don’t use wx.Parse can you remove it from the PR?

Can you also update the readme with npm run build and copy /dist/parse.weapp.js

**Comment by einsqing** [comment]
@dplewis Update readme and remove wx.Parse

**Pull request merged**